### ISSUE TITLE
Error: INVALID_POINTER/Segmentation Fault

### ISSUE THREAD

**Issue by pmdevita** [ImageOptim/gifski#22]
Stumbled across this while working on a reddit bot that reverses gifs.

Took the mp4 of an imgur gif https://i.imgur.com/sG7sgqZ.mp4

Created pngs using ffmpeg with the command `ffmpeg -i in.mp4 frame%04d.png`

Then ran gifski `gifski -o temp.gif frame*.png`

Different runs have had different results. Sometimes it sits there and then comes back with either `Segmentation Fault` or `Error: INVALID_POINTER`. Other times it gets about two frames into the process before giving one of the errors.

I'm using Debian 9 with ffmpeg version N-89805-gcbbb206 and gifski 0.7.1

**Comment by kornelski** [comment]
Can you post full console output?

**Comment by kornelski** [comment]
If you could get a backtrace, that'd be even more useful:

1. Get Rust/Cargo
2. Clone source
3. `cargo build`
4. `lldb -- target/debug/gifski -o temp.gif frame*.png`
5. `r`
6. `bt`

**Comment by pmdevita** [comment]
Forgot to mention I was using the binary from the releases page. 

It completed fine without error. So I went ahead and built the release and that also worked fine. I guess the last commit fixed it? Thanks for your help!

**Comment by kornelski** [comment]
OK, that's a tough one. It looks like OpenMP version corrupts memory all over the place, causing errors in unrelated places :(

**Comment by phw** [comment]
I had experienced crashes with openmp enabled, too, when I packaged up gifski in Snap and Flatpak packages. Gifski would just segfault at some point, sometimes immediatelly sometimes early in the processing. I never came around to report this since I could not find the time to properly debug this inside the containers, but this looks very similar.

I have no issues with openmp enabled on my local box (Arch), but maybe it is something Debian specific? At least the Snap runtime is also Ubuntu based, not sure about Flatpak.
